packages/web3-eth/src/eth_tx.ts

```diff
@@ -279,17 +263,23 @@ export const prepareTransactionForSigning = async (
 	web3Context: Web3Context<EthExecutionAPI>,
 	privateKey?: HexString | Buffer,
 ) => {
-	const populatedTransaction = await populateTransaction(
-		transaction,
+	const transactionBuilder = web3Context.transactionBuilder ?? defaultTransactionBuilder;
+
+	const populatedTransaction = await transactionBuilder({
+		transaction: transaction as unknown as Record<string, unknown>,
 		web3Context,
-		ValidTypes.HexString,
 		privateKey,
-	);
+	});
+
+	const formattedTransaction = formatTransaction(
+		populatedTransaction,
+		ValidTypes.HexString,
+	) as PopulatedUnsignedTransaction<HexString>;
```

Comment:
No matter who build the transaciton, we have to format it properly before signing, so why I extracted formatting part to `prepareTransactionForSigning`